### PR TITLE
CNTRLPLANE-3685: docs - add Agentic SDLC context files

### PR DESCRIPTION
**Pull request by mehabhalodiya**
This PR adds the documentation context files identified in the Shift Week Contextification initiative:

- CONTRIBUTING.md: Development workflow, PR process, testing requirements
- AGENTS.md: AI agent guidance for working with this codebase
- ARCHITECTURE.md: Design decisions, component architecture, tradeoffs

These files provide prescriptive context to help both humans and AI agents understand how to contribute effectively to kube-rbac-proxy.

Jira: [CNTRLPLANE-3685](https://redhat.atlassian.net/browse/CNTRLPLANE-3685)

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive guidance on project architecture, security practices, development workflows, testing, contribution standards, and downstream maintenance.
  * Documented request processing, authentication and authorization flows, proxy behavior, integrations, performance considerations, and testing strategies.
  * Expanded onboarding information for local setup, CI checks, generated files, dependency updates, release processes, and contribution requirements.
  * Updated the main agent guidance to reference the downstream instructions.

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+2. **TLS First**: Prefer mTLS for authentication over bearer tokens for better security properties.
+
+3. **Authorization Model**: The proxy performs a SubjectAccessReview against the Kubernetes API for every request. This is the core security mechanism - do not bypass or weaken it.
+
+### Project Status
+
+- **Alpha stage**: Breaking changes are expected
+- **Deprecation path**: Some flags/features are being deprecated as the project aligns with Kubernetes standards
+- **Feature branch**: `sig-auth-acceptance` shows planned changes for Kubernetes project acceptance
```
Comment (ehearne-redhat) [inline AGENTS.md]: I don't think we see this in the openshift fork.

File: DOWNSTREAM_AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+### Security Considerations
+
+1. **Token Security**: This proxy can receive bearer tokens. The receiving side can use these tokens to impersonate the client. Only use token authentication when the receiving side is already higher-privileged or the token is low-privileged. See [README - ServiceAccount token security](README.md#notes-on-serviceaccount-token-security).
+
+2. **TLS First**: Prefer mTLS for authentication over bearer tokens for better security properties.
+
+3. **Authorization Model**: The proxy performs a SubjectAccessReview against the Kubernetes API for every request. This is the core security mechanism - do not bypass or weaken it.
+
+### Project Status
```
Comment (ehearne-redhat) [inline DOWNSTREAM_AGENTS.md:39]: I'm unsure about this section because I feel it bases itself from upstream. We do use kube-rbac-proxy a lot in OpenShift.

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+### Common Pitfalls
+
+1. **Do not add flags for insecure operation**: The project is deprecating `--insecure-listen-address` and similar flags. New changes should not add insecure modes.
+
+2. **License headers are required**: All Go files must have license headers. Run `make check-license` before submitting changes.
+
+3. **Generated files**: The README includes embedded help text and examples. After changing flags or examples, run `make generate` and commit the changes.
+
+4. **Kind cluster override**: `make test-local` creates/recreates the default kind cluster. This will destroy any existing default kind cluster.
```
Comment (ehearne-redhat) [inline AGENTS.md]: Nice catch but maybe we should mention it does this to run tests locally?

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+5. **Test timeouts**: E2E tests have a 55-minute timeout. Keep tests focused and efficient.
```
Comment (ehearne-redhat) [inline AGENTS.md]: Looking at [go test official docs](https://pkg.go.dev/cmd/go/internal/test#:~:text=%2Dtimeout%20d%0A%09%20%20%20%20If%20a%20test%20binary%20runs%20longer%20than%20duration%20d%2C%20panic.%0A%09%20%20%20%20If%20d%20is%200%2C%20the%20timeout%20is%20disabled.%0A%09%20%20%20%20The%20default%20is%2010%20minutes%20(10m).) default test timeout is 10 minutes. So having it set to 55 minutes is 5.5x more than the standard. Checking commit history leads to https://github.com/kube-rbac-proxy/kube-rbac-proxy/pull/38 and no discussion as to why 55 minutes was set. 

What I'm seeing [here](https://github.com/kube-rbac-proxy/kube-rbac-proxy/blob/master/.github/workflows/build.yml#L116-L119 ) is even more contradictory - 30 minute timeout set for github actions e2e-test workflow run. 

I would consider removing this line as the 55 minutes appear to be an arbitrary number with no current links to a purpose.

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+golangci-lint run   # Linting (configured in .golangci.yaml)
+make test-unit      # Unit tests
+make test-e2e       # E2E tests (requires kind cluster)
+```
+
+## Key Design Patterns
+
+### Proxy Architecture
+
+The proxy sits between a client and an upstream service:
```
Comment (ehearne-redhat) [inline AGENTS.md]: I'm unsure about stating its an upstream service since we use it in OpenShift here. Technically speaking it does sit behind Prometheus but it is our own version.  It wouldn't be a bad idea to give an example of service it sits behind too.

File: AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+```
+Client -> kube-rbac-proxy -> Upstream Service
+          |
+          v
+     Kubernetes API (SubjectAccessReview)
+```
+
+1. **Authentication**: Validate client identity via client TLS cert or bearer token (TokenReview against k8s API)
+2. **Authorization**: Perform SubjectAccessReview against k8s API
+3. **Proxy**: If authorized, forward request to upstream
```
Comment (ehearne-redhat) [inline AGENTS.md]: I think this should say 'forward request to client' ?

File: AGENTS.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+### Configuration Model
+
+- **CLI flags**: Most configuration via command-line flags (see `--help`)
```
Comment (ehearne-redhat) [inline AGENTS.md]: Should we link it to `cmd/kube-rbac-proxy/app/options/options.go` under `Flags()` ? Because maybe we did not build the binary yet and the agent wants to learn about flags for contributing.

File: DOWNSTREAM_AGENTS.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+1. **Authentication**: Validate client identity via client TLS cert or bearer token (TokenReview against k8s API)
+2. **Authorization**: Perform SubjectAccessReview against k8s API
+3. **Proxy**: If authorized, forward request to upstream
+
+### Configuration Model
+
+- **CLI flags**: Most configuration via command-line flags (see `--help`)
+- **Config file**: `--config-file` for SubjectAccessReview details and rewrites
```
Comment (ehearne-redhat) [inline DOWNSTREAM_AGENTS.md:121]: Same idea here, but I also think we should provide a minimal working config file as context to the agent.

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+
+- Supports HTTP/1.1, HTTP/2, and h2c (HTTP/2 cleartext)
+- Can use client certificates for upstream authentication (`--upstream-client-cert-file`)
+- Supports custom CA for upstream TLS verification (`--upstream-ca-file`)
+
+## Common Operations
+
+### Adding a New Flag
+
+1. Add flag definition in `cmd/kube-rbac-proxy/main.go`
```
Comment (ehearne-redhat) [inline AGENTS.md]: Sadly this workflow is wrong. We need to do this in `cmd/kube-rbac-proxy/cmd/options/options.go` . 

Steps 1 + 2 can be combined as when we inherently define a flag, we give help text too. 

Example: 

```golang
flagset.StringVar(&o.InsecureListenAddress, "insecure-listen-address", "", "[DEPRECATED] The address the kube-rbac-proxy HTTP server should listen on.")
```

File: AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+2. Update help text and any affected configuration structs
+3. Run `make generate` to update embedded documentation in README
```
Comment (ehearne-redhat) [inline AGENTS.md]: I'm unsure about this step because looking at [the script it uses](https://github.com/openshift/kube-rbac-proxy/blob/d12e274605248f6c59373240a7eae7a7a357dcb3/scripts/generate-examples.sh), it just seems to update current yaml examples with the latest version of kube-rbac-proxy. It also just seems to run `kube-rbac-proxy -h` and output that to `_output/help.txt` [which is gitignored](https://github.com/openshift/kube-rbac-proxy/blob/d12e274605248f6c59373240a7eae7a7a357dcb3/.gitignore) sadly. 

I would suggest removing this step as it doesn't lead to the suggested outcome.

File: AGENTS.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+4. Add unit tests for flag parsing
+5. Add e2e tests if the flag affects runtime behavior
```
Comment (ehearne-redhat) [inline AGENTS.md]: It seems current testing culture is to minimally test flags being set. 

https://github.com/openshift/kube-rbac-proxy/blob/d12e274605248f6c59373240a7eae7a7a357dcb3/test/e2e/basics.go#L99

Maybe we can drop step 4 here because we don't have unit tests for flag parsing, it seems.

We might also want to direct to available test locations and convention too.

File: AGENTS.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+### Modifying Authentication/Authorization
+
+Changes to `pkg/filters/` or `pkg/hardcodedauthorizer/` are security-critical:
```
Comment (ehearne-redhat) [inline AGENTS.md]: Why are changes to these two directories security-critical in particular? 

It wouldn't be a bad idea to elaborate further on the purpose of these two directories.

File: AGENTS.md [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+
+1. Understand the current security model first
```
Comment (ehearne-redhat) [inline AGENTS.md]: What does 'security model' mean in this case? If I can't follow it, I am not confident that the agent could either.

File: AGENTS.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+2. Write comprehensive unit tests covering edge cases
```
Comment (ehearne-redhat) [inline AGENTS.md]: I'm not seeing this approach currently in the repo. Writing minimal tests seems to be the way here.

File: AGENTS.md [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+3. Add e2e tests demonstrating the change in a real cluster
```
Comment (ehearne-redhat) [inline AGENTS.md]: I think this is a valid step. However, we probably don't need to mention `in a real cluster` .

File: AGENTS.md [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+4. Document security implications in PR description
```
Comment (ehearne-redhat) [inline AGENTS.md]: I don't think we should set the expectation that the agent needs to create a PR. Maybe the user would rather do that part. If we want to have a section specifically documenting expectations for a PR, we can mention the workflow there.

File: AGENTS.md [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+5. Expect thorough review from security-focused maintainers
```
Comment (ehearne-redhat) [inline AGENTS.md]: I'm trying to understand why an agent would need to know this - what would we expect the agent to achieve with expecting a thorough review? Maybe we should be more verbose on what a reviewer would expect from the changes, and that could go into a `Opening a PR` section if needed. 

I don't think it is a necessary step to include.

**Comment by ehearne-redhat** [comment]
Still going through the review process - but as an aside, what do you think about adding a `README-openshift.md` file where we can provide more in-depth explanation of the repo from our/the agent's perspective? Do you think there is a need for that context for the agent/us?

File: ARCHITECTURE.md [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+**Reduced Surface**:
+- Proxy only forwards to one upstream (no routing complexity to exploit)
+- All requests are authenticated and authorized before forwarding
+- No dynamic configuration or admin API (smaller attack surface)
+
+## Future Direction
+
+### Kubernetes SIG-Auth Acceptance
+
+The project is working toward becoming an official Kubernetes project under SIG Auth. This involves:
```
Comment (ehearne-redhat) [inline ARCHITECTURE.md]: Yes, this is true, but be sure to clarify the upstream of the repo is working towards that.

File: ARCHITECTURE.md [hunk 19]
```diff
@@ -0,0 +1,326 @@
...
+
+- **Removing insecure modes**: Deprecating `--insecure-listen-address` and similar flags
+- **Aligning with Kubernetes logging**: Removing deprecated logging flags
+- **Tighter integration with Kubernetes code**: Using more upstream Kubernetes libraries
+
+See `sig-auth-acceptance` feature branch for planned changes.
```
Comment (ehearne-redhat) [inline ARCHITECTURE.md]: As mentioned before, we don't have this branch on this fork. :)

File: CLAUDE.md [hunk 20]
```diff
@@ -0,0 +1 @@
+AGENTS.md
```
Comment (ehearne-redhat) [inline CLAUDE.md]: This should probably be `@AGENTS.md` for claude agent specifically.

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,191 @@
...
+- Run unit tests: `make test-unit`
+- Unit tests should be fast and not require external dependencies
+- Tests must pass with race detection enabled (`-race` flag)
+
+#### End-to-End Tests
+
+- Required for changes affecting core proxy behavior
+- Run e2e tests locally: `make test-local` (sets up kind cluster and runs full test suite)
+- E2E tests run in a kind cluster and validate real Kubernetes integration
+- Test timeout: 55 minutes maximum
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: As mentioned above we probably want to remove this as it seems to be arbitrary.

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,191 @@
...
+  - License header check
+  - Go linting
+  - Unit tests
+  - E2E tests
+  - Generate check (ensures generated docs/examples are current)
+  - Build check
+
+### Code Review Expectations
+
+- **Turnaround time**: Reviewers aim to provide initial feedback within 1-2 business days
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: Is this necessarily the case?

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,191 @@
...
+- **What reviewers look for**:
+  - Correctness and security implications
+  - Test coverage and quality
+  - Code clarity and maintainability
+  - Alignment with project goals and architecture
+  - Breaking changes or deprecation path
+
+### Merge Requirements
+
+- Minimum 1 approval from an OWNER
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: We should probably delve into tide requirements here too. You will need `lgtm` and `approved` for example.

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,191 @@
...
+
+## Project Status and Direction
+
+**Current Status**: This project is in **alpha** stage. Flags, configuration, behavior, and design may change significantly in future releases.
+
+**Future Direction**: The project is working toward acceptance as a Kubernetes project. As part of this:
+
+- We're aligning more tightly with Kubernetes code and conventions
+- Some flags/features are being deprecated (e.g., insecure listening, certain logging flags)
+- A `sig-auth-acceptance` feature branch shows planned changes
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: We'll probably also want to remove this too.

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,191 @@
...
+
+## Getting Help
+
+- **Questions about contributing?** Open a GitHub issue with the `question` label
+- **Bug reports**: Open an issue with steps to reproduce
+- **Feature requests**: Open an issue describing the use case and proposed solution
+
+## Code of Conduct
+
+This project follows the Kubernetes community [Code of Conduct](https://kubernetes.io/community/code-of-conduct/).
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: Not necessarily, as this is our fork.

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,191 @@
...
+- Some flags/features are being deprecated (e.g., insecure listening, certain logging flags)
+- A `sig-auth-acceptance` feature branch shows planned changes
+
+See the [README](README.md#current-future-and-deprecation-of-flags--features) for more details.
+
+## Release Process
+
+Releases are handled by project maintainers. See [RELEASE.md](RELEASE.md) for the release process documentation.
+
+## Getting Help
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: Some mention of internal slack channels for red hat employees wouldn't be a bad idea.

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,191 @@
...
+
+- **Questions about contributing?** Open a GitHub issue with the `question` label
+- **Bug reports**: Open an issue with steps to reproduce
+- **Feature requests**: Open an issue describing the use case and proposed solution
```
Comment (ehearne-redhat) [inline CONTRIBUTING.md]: I believe this doesn't apply to our fork.

File: AGENTS.md (hunk 5 above)
Comment (everettraven) [reply AGENTS.md]: I think that "upstream service" here means a literal service that the proxy forwards request to (i.e the service is "upstream" of the proxy in the HTTP request flow)

File: AGENTS.md (hunk 6 above)
Comment (everettraven) [reply AGENTS.md]: Proxies don't forward a request to a client - they forward them to another service.

File: AGENTS.md (hunk 10 above)
Comment (everettraven) [reply AGENTS.md]: It looks like it uses https://github.com/campoy/embedmd and references the `_output/help.txt` file to regenerate the help section in the readme.

See: https://github.com/openshift/kube-rbac-proxy/blob/d12e274605248f6c59373240a7eae7a7a357dcb3/README.md?plain=1#L46

File: DOWNSTREAM_AGENTS.md
Comment (everettraven) [inline DOWNSTREAM_AGENTS.md:1]: Because this is a fork of an upstream project, should we call this `DOWNSTREAM_AGENTS.md` to prevent a future clash if the upstream kube-rbac-proxy adds an `AGENTS.md` file in the future?

File: DOWNSTREAM_ARCHITECTURE.md [hunk 28]
```diff
@@ -0,0 +1,326 @@
...
+
+1. **Single Upstream**: One kube-rbac-proxy instance protects exactly one upstream. This keeps the configuration simple and the security boundary clear.
+
+2. **Kubernetes-Native Authorization**: Leverage Kubernetes' existing RBAC system rather than implementing a separate authorization mechanism. Every request is validated against the Kubernetes API server.
+
+3. **Defense in Depth**: Even in clusters without NetworkPolicies, or for HostNetwork pods where NetworkPolicies don't apply, we can enforce access control.
+
+4. **Sidecar Pattern**: Designed to run as a sidecar container alongside the service being protected.
+
+## Architecture Diagram
```
Comment (everettraven) [inline DOWNSTREAM_ARCHITECTURE.md:19]: For architecture diagrams, could we use the mermaid format instead?

See https://mermaid.js.org/ for more information on mermaid diagrams.

GitHub has support for rendering them by setting the language on the fenced code block to `mermaid` and they render significantly nicer than ASCII diagrams.

File: DOWNSTREAM_ARCHITECTURE.md
Comment (everettraven) [inline DOWNSTREAM_ARCHITECTURE.md:1]: Similar comment on file naming.

File: CONTRIBUTING.md
Comment (everettraven) [inline CONTRIBUTING.md:1]: Similar file naming comment.

Comment (everettraven) [inline CONTRIBUTING.md:1]: This looks like a pretty good contributing guide for the upstream project, but not necessarily for OpenShift.

Some things I think that would be important here for us to mention specifically for OpenShift:
- For non-control-plane team members, how should they request reviews for changes they are contributing?
- For control-plane team members, how should they request reviews for changes they are contributing?
- How do we sync changes from the upstream project to our fork?
- How should commits that are specific to OpenShift be made?

https://docs.google.com/document/d/14-MqX0qoZeTO0X16b7sGQ5--lvAUaybLH5erQu-f3zk/edit?usp=sharing might be a better baseline for an OpenShift-specific contribution guide.

File: AGENTS.md (hunk 1 above)
Comment (mehabhalodiya) [reply AGENTS.md]: - Removed "Alpha stage" (not accurate for OpenShift production use)
- Removed sig-auth-acceptance reference (doesn't apply to OpenShift fork - already fixed)
- Added production use context to reflect actual OpenShift usage

File: AGENTS.md (hunk 3 above)
Comment (mehabhalodiya) [reply AGENTS.md]: Added clarification about purpose (running tests locally)

File: AGENTS.md (hunk 4 above)
Comment (mehabhalodiya) [reply AGENTS.md]: Removed to avoid confusion

File: AGENTS.md (hunk 5 above)
Comment (mehabhalodiya) [reply AGENTS.md]: - Changed "Upstream Service" to "Backend Service" to avoid confusion with "upstream" project
- Added example "(e.g., Prometheus /metrics)" to clarify what kind of service

File: AGENTS.md (hunk 6 above)
Comment (mehabhalodiya) [reply AGENTS.md]: Terminology updated throughout to use "Backend Service" for consistency

File: AGENTS.md (hunk 7 above)
Comment (mehabhalodiya) [reply AGENTS.md]: link to `cmd/kube-rbac-proxy/app/options/options.go` with explanation that flags are defined in the Flags() method

File: DOWNSTREAM_AGENTS.md (hunk 8 above)
Comment (mehabhalodiya) [reply DOWNSTREAM_AGENTS.md:121]: Added inline example config:
```
authorization:
  resourceAttributes:
    namespace: default
    apiVersion: v1
    resource: services
    name: my-service
```
Plus reference to `examples/` directory for complete examples.

File: AGENTS.md (hunk 12 above)
Comment (mehabhalodiya) [reply AGENTS.md]: Explained what each directory does and why it's security-critical.

File: AGENTS.md (hunk 13 above)
Comment (mehabhalodiya) [reply AGENTS.md]: Added more concrete language with reference to where to learn more.

File: CONTRIBUTING.md (hunk 25 above)
Comment (mehabhalodiya) [reply CONTRIBUTING.md]: Kept it - while this is an OpenShift fork, following Kubernetes CoC is still appropriate and beneficial. : )

File: CLAUDE.md [hunk 29]
```diff
@@ -0,0 +1 @@
+DOWNSTREAM_AGENTS.md
```
Comment (ehearne-redhat) [inline CLAUDE.md:1]:
```suggestion
@DOWNSTREAM_AGENTS.md
```

File: DOWNSTREAM_CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,262 @@
...
+1. Fork and clone the repository:
+   ```bash
+   git clone https://github.com/YOUR-USERNAME/kube-rbac-proxy.git
+   cd kube-rbac-proxy
+   ```
+
+2. Add the upstream OpenShift remote:
+   ```bash
+   git remote add upstream https://github.com/openshift/kube-rbac-proxy.git
+   ```
```
[the remark this answers is not in the document]
Comment (ehearne-redhat) [reply DOWNSTREAM_CONTRIBUTING.md:29]: The above approach is correct. It might not be a bad idea to show how to add the original. So the remotes would be `your-fork` --> `upstream` (this repo) --> `original` (brancz original repo)

File: DOWNSTREAM_CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,262 @@
+# Contributing to OpenShift kube-rbac-proxy
+
+Thank you for your interest in contributing to the OpenShift fork of kube-rbac-proxy! This document provides guidelines specific to contributing to the OpenShift downstream version.
+
+## OpenShift Fork Context
+
+This repository is the OpenShift downstream fork of kube-rbac-proxy, maintained for use in OpenShift clusters. We sync changes from the upstream brancz/kube-rbac-proxy project while maintaining OpenShift-specific requirements and integrations.
+
```
Comment (ehearne-redhat) [inline DOWNSTREAM_CONTRIBUTING.md:8]: Should we emphasise a upstream-first approach ? So, if the change can be made in `brancz/kube-rbac-proxy` then it should and then it should be cherry-picked here. Then, downstream-specific contributions can be done here e.g. customer request that would not have benefit upstream.

File: DOWNSTREAM_CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,262 @@
...
+### Syncing from Upstream
+
+The Control Plane team periodically syncs changes from upstream (brancz/kube-rbac-proxy). The process:
+
+1. **Upstream tracking**: Monitor upstream releases and relevant PRs
+2. **Cherry-pick or merge**: Bring in upstream changes, resolving conflicts with OpenShift-specific patches
+3. **Testing**: Run full test suite to ensure upstream changes work with OpenShift
+4. **Documentation**: Update CHANGELOG.md noting upstream version synced
+
+**For contributors**: If you're implementing a feature that would benefit upstream, consider:
```
Comment (ehearne-redhat) [inline DOWNSTREAM_CONTRIBUTING.md:197]: I like this section - could we reference it near the top of the document, maybe hyperlink?

**Review by ehearne-redhat — COMMENTED** [review]
Overall, the changes look good. Just a few minor changes. :)

File: DOWNSTREAM_CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,267 @@
...
+See [Prow Command Help](https://prow.ci.openshift.org/command-help) for full list.
+
+## OpenShift-Specific Contributions
+
+### Marking Downstream-Only Changes
+
+When making changes specific to OpenShift (not from upstream), include `[DOWNSTREAM]` in the commit message:
+
+```text
+[DOWNSTREAM] Add OpenShift-specific TLS cipher configuration
```
Comment (ehearne-redhat) [inline DOWNSTREAM_CONTRIBUTING.md]: Apologies but I don't believe I caught this one - I'm not seeing this convention in the repo commits. Was this something you saw in commit messages here?

**Review by ehearne-redhat — COMMENTED** [review]
New changes look good but I did catch something about commit message conventions that should probably be addressed.

File: DOWNSTREAM_CONTRIBUTING.md (hunk 33 above)
Comment (mehabhalodiya) [reply DOWNSTREAM_CONTRIBUTING.md]: Good catch — you're right, there's no `[DOWNSTREAM]` prefix convention in this repo (I confirmed zero commits use it). I've removed it everywhere and replaced it with the actual pattern: descriptive commit messages, with downstream syncs landing via `*-downstream` merge branches. Thanks!

**Comment by ehearne-redhat** [comment]
/lgtm
/verified by ehearne-redhat

@mehabhalodiya you'll likely need an approve from @everettraven and update the jira ticket to resolve invalid reference. 

Nice job on getting this done. 🥳
